compiler/luci/pass/src/CircleOptimizerUtils.cpp

```diff
@@ -97,4 +98,23 @@ bool has_dynamic_shape(const loco::Node *node)
   return false;
 }
 
+std::vector<loco::Node *> alive_nodes(loco::Graph *g)
+{
+  auto result = loco::postorder_traversal(loco::output_nodes(g));
+
+  for (auto node : loco::all_nodes(g))
+  {
+    // already checked as alive
+    if (std::find(result.begin(), result.end(), node) != result.end())
+      continue;
+
+    // This node is not used for both graph output and multiple output operation
+    if (node->dialect()->service<DeadNodeQueryServiceImpl>()->isDeadNode(node))
+      continue;
+
+    result.emplace_back(node);
+  }
+  return result;
+}
```

Comment:
I don't think this is the right place for this method.
Please add a brief description of this method.

Reply:
Then I will introduce `luci/IR/Algorithm.h` and define this function there like `loco/IR/Algorithm.h`.
Adding a brief description will be handled there too.